x-pack/plugins/lists/server/services/extension_points/extension_point_storage_client.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from 'kibana/server';
+
+import { ExtensionPoint, NarrowExtensionPointToType } from './types';
+import { ExtensionPointError } from './errors';
+import { ExtensionPointStorageInterface } from './extension_point_storage';
+
+export class ExtensionPointStorageClient {
+  constructor(
+    private readonly storage: ExtensionPointStorageInterface,
+    private readonly logger?: Logger
```

Comment:
I would replace this with: `private readonly logger?: Logger` and just make passing down a logger required. I don't see any reason to make the logger optional. You will have to update a test or two to push down a mock logger but the less optional components such as the logger, the better off we will be with maintainers and others within this section of the code.

Also, most of the other parts of the code all require a logger to passed down, so this feels like it goes against the other patterns of just requiring the logger. Also, it won't put a burden on any dependent functions or methods to always have to define their loggers as `logger | undefined` or `logger?` if we just require it.

Really in a lot of our codebase we do a bad pattern of making things optional when they're always being used and passed down which just adds to the `?` noise or branching logic and extra types for `undefined` that aren't needed.

IMHO, the less we have to deal with `undefined` and `null` the better off we are.

Reply:
Cool. I'll make the change to have it as required